### PR TITLE
Webpack/Babel setup

### PR DESCRIPTION
**Pull request by infiniteluke**
_Please ignore the branch names.. Instead of having a `rewrite` branch we merge into, we can just use this 4K fork `master` as the new frontend_
## Test
- Make sure `npm run build` and `npm start` do what they're supposed to.
- Browse to http://localhost:8080/webpack-dev-server/ and make sure breakpoint mixin is working by adjusting viewport.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "es2015",
+    "stage-0",
+    "stage-1",
+    "stage-2",
+    "stage-3"
+  ],
+  "ignore": [
+    "node_modules"
+  ]
+}
```
Comment (flipactual) [inline .babelrc:10]: Nuh uh

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,4 @@ npm-debug.log
 # Sublime Text project files:
 *.sublime-*
```
Comment (flipactual) [inline .gitignore]: What is this about? Why do we need editor specific project files?

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import { render } from 'react-dom';
+import '!style!css!sass!./scss/main.scss';
```
Comment (flipactual) [inline src/index.js]: This can/should be specific in webpack configuration.

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  plugins: [
+    new HtmlWebpackPlugin({
+      template: 'src/index.html',
+      inject: 'body',
+      filename: 'index.html',
+    }),
+  ],
+};
+
+module.exports = config;
```
Comment (flipactual) [inline webpack.config.js]: This should be on line 3

File: package.json [hunk 5]
```diff
@@ -1,18 +1,12 @@
...
   "scripts": {
-    "start": "node scrum_cards_server.js"
-  },
-  "dependencies": {
-    "npm": "~1.1.71",
-    "express": "~3.1.2",
-    "socket.io": "~0.9.16",
-    "yargs": "~1.2.1"
-  },
-  "engines": {
```
Comment (flipactual) [inline package.json:14]: Can we lock into the same engine as the backend?

File: .babelrc [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  "presets": [
+    "react",
+    "es2015",
+    "stage-0",
+    "stage-1",
+    "stage-2",
+    "stage-3"
+  ],
+  "ignore": [
+    "node_modules"
```
Comment (flipactual) [inline .babelrc]: Is this not automatic?

File: .eslintignore [hunk 7]
```diff
@@ -0,0 +1 @@
+dist
```
Comment (flipactual) [inline .eslintignore:1]: There's more of these.

**Comment by infiniteluke** [comment]
Using precss with the postcss-scss loader doesn't work well apparently. So we'll just use the sass laoder for now.. https://github.com/jonathantneal/precss/issues/49#issuecomment-169900817

File: .eslintignore (hunk 7 above)
Comment (flipactual) [inline .eslintignore:1]: Still need to address the lack of newlines

File: .eslintrc [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "env": {
+    "es6": true,
+    "node": true
+  },
+  "extends": "airbnb"
+}
```
Comment (flipactual) [inline .eslintrc:7]: We should convert this to `.eslintrc.yml`... can address later, but may as well make that change when adding the newline.

Comment (flipactual) [inline .eslintrc:7]: Just if you want to

File: src/index.html [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+  <head>
+    <title>Scrummy - By Four Kitchens</title>
+    <meta charset="utf-8" />
+  </head>
+  <body>
+    <main>
+      <noscript>You cannot play Scrummy without JavaScript.</noscript>
+    </main>
+  </body>
+</html>
```
Comment (flipactual) [inline src/index.html:11]: Needs newline.

File: src/scss/main.scss [hunk 10]
```diff
@@ -0,0 +1,10 @@
+@import "~breakpoint-sass/stylesheets/breakpoint";
+@import "~singularitygs/stylesheets/singularitygs";
+@import "~sass-toolkit/stylesheets/toolkit/kickstart";
+
+main {
+  display: flex;
+  @include breakpoint(500px) {
+    color: blue;
+  }
+}
```
Comment (flipactual) [inline src/scss/main.scss:10]: Needs newline.

File: .babelrc [hunk 11]
```diff
@@ -0,0 +1,10 @@
+{
+  "presets": [
+    "react",
+    "es2015",
+    "stage-0",
+    "stage-1",
+    "stage-2",
+    "stage-3"
+  ]
+}
```
Comment (flipactual) [inline .babelrc:10]: Needs newline.

**Comment by infiniteluke** [comment]
It has been decided that scrummy's front end will be a separate repo.

**Pull request closed without merging**